### PR TITLE
HashProto: Use fastssz when available

### PR DESCRIPTION
**Pull request by prestonvanloon**
Before
```
BenchmarkHashProto-8             1596458               829 ns/op             144 B/op          2 allocs/op
```
After (if the struct meets the interface)
```
BenchmarkHashProto-8             3118893               369 ns/op             296 B/op          4 allocs/op
```

### CODE REVIEW

**Comment by prestonvanloon** [comment]
This needs to go into v0.11 because we use hashProto as a database key for some objects

**Comment by rauljordan** [comment]
Looks like some tests fail @prestonvanloon

**Comment by prestonvanloon** [comment]
Working on tests now. Some test input is invalid now against ssz types.

I'm thinking I want to merge this to master instead of v0.11.
Looking into using the db migration bucket to toggle this.

File: beacon-chain/cache/checkpoint_state_test.go [hunk 1]
```diff
@@ -4,14 +4,16 @@ import (
 	"reflect"
 	"testing"
 
+	"github.com/prysmaticlabs/prysm/shared/bytesutil"
+
```
Comment (terencechain) [inline beacon-chain/cache/checkpoint_state_test.go]:
```suggestion
```

**Pull request merged**